exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+const (
+	namespace          = "otel"
+	otelTraceID        = namespace + ".trace_id"
+	otelSpanID         = namespace + ".span_id"
+	otelSeverityNumber = namespace + ".severity_number"
+	otelSeverityText   = namespace + ".severity_text"
+	otelTimestamp      = namespace + ".timestamp"
+
+	ddStatus    = "status"
+	ddTraceID   = "dd.trace_id"
+	ddSpanID    = "dd.span_id"
+	ddTimestamp = "@timestamp"
+)
+
+const (
+	logLevelTrace = "trace"
+	logLevelDebug = "debug"
+	logLevelInfo  = "info"
+	logLevelWarn  = "warn"
+	logLevelError = "error"
+	logLevelFatal = "fatal"
+)
+
+// Transform is responsible to convert LogRecord to datadog format
+func Transform(lr plog.LogRecord, res pcommon.Resource, sendLogRecordBody bool) datadogV2.HTTPLogItem {
+	hostName, serviceName := extractHostNameAndServiceName(res.Attributes(), lr.Attributes())
+
+	l := datadogV2.HTTPLogItem{
+		AdditionalProperties: make(map[string]string),
+	}
+	if hostName != "" {
+		l.Hostname = datadog.PtrString(hostName)
+	}
+
+	if serviceName != "" {
+		l.Service = datadog.PtrString(serviceName)
+	}
+
+	// we need to set log attributes as AdditionalProperties
+	lr.Attributes().Range(func(k string, v pcommon.Value) bool {
+		l.AdditionalProperties[k] = v.AsString()
+		return true
+	})
+	if !lr.TraceID().IsEmpty() {
+		l.AdditionalProperties[ddTraceID] = strconv.FormatUint(traceIDToUint64(lr.TraceID()), 10)
+		l.AdditionalProperties[otelTraceID] = lr.TraceID().HexString()
+	}
+	if !lr.SpanID().IsEmpty() {
+		l.AdditionalProperties[ddSpanID] = strconv.FormatUint(spanIDToUint64(lr.SpanID()), 10)
+		l.AdditionalProperties[otelSpanID] = lr.SpanID().HexString()
+	}
+	var status string
+
+	// we want to use the serverity that client has set on the log and let datadog backend
+	// decide the appropriate level
+	if lr.SeverityText() != "" {
+		status = lr.SeverityText()
+		l.AdditionalProperties[otelSeverityText] = lr.SeverityText()
+	} else if lr.SeverityNumber() != 0 {
+		status = derviveStatusFromSeverityNumber(lr.SeverityNumber())
+	}
+
+	l.AdditionalProperties[ddStatus] = status
+	// if SeverityNumber is set , we want to retain it
+	if lr.SeverityNumber() != 0 {
+		l.AdditionalProperties[otelSeverityNumber] = strconv.Itoa(int(lr.SeverityNumber()))
+	}
+
+	// for datadog to use the same timestamp we need to set the additional property of "@timestamp"
+	if lr.Timestamp() != 0 {
+		// we are retaining the nano second precision in this property
+		l.AdditionalProperties[otelTimestamp] = strconv.FormatInt(lr.Timestamp().AsTime().UnixNano(), 10)
+		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format(time.RFC3339)
+	}
+
+	if sendLogRecordBody {
+		// set the Message
+		l.Message = lr.Body().AsString()
+	}
+
```

Comment:
```suggestion
	}
	// for datadog to use the same timestamp we need to set the additional property of "@timestamp"
	if lr.Timestamp() != 0 {
		// we are retaining the nano second precision in this property
		l.AdditionalProperties[otelTimestamp] = strconv.FormatInt(lr.Timestamp().AsTime().UnixNano(), 10)
		l.AdditionalProperties[ddTimestamp] = lr.Timestamp().AsTime().Format(time.RFC3339)
	}
	if sendLogRecordBody {
		// set the Message
		l.Message = lr.Body().AsString()
	}
```
I don't think there are any reasons to start new paragraphs in the code.